src/service/vsync-impl.js

```diff
@@ -108,8 +108,27 @@ export class Vsync {
     /** @const {!Function} */
     this.boundRunScheduledTasks_ = this.runScheduledTasks_.bind(this);
 
-    /** @const {!Pass} */
-    this.pass_ = new Pass(this.win, this.boundRunScheduledTasks_, FRAME_TIME);
+    /**
+     * If the doc is invisible we use this instead of rAF because rAF
+     * does not run in that scenario.
+     * However, we only do this for non-animation tasks as running
+     * animations doesn't make sense when not visible.
+     * @const {!Pass}
+     */
+    this.invisiblePass_ = new Pass(this.win, this.boundRunScheduledTasks_,
+        FRAME_TIME);
+
+    /**
+     * Similar to this.invisiblePass_, but backing up a real rAF call. If we
+     * somehow failed to know that we are throttled we use a timer (which
+     * may also be throttled but at least runs eventually) to make sure
+     * we continue to get work done.
+     * @const {!Pass}
+     */
+    this.backupPass_ = new Pass(this.win, this.boundRunScheduledTasks_,
```

Comment:
We always schedule this and an rAF and cancel the pass when the rAF runs. Under normal rAF operation I do not see how this could ever not be cancelled.